src/vm/compile.cpp

```diff
@@ -1007,13 +1007,19 @@ void CEECompileInfo::GetCallRefMap(CORINFO_METHOD_HANDLE hMethod, GCRefMapBuilde
 
     UINT nStackSlots;
 
+#ifdef _TARGET_ARM64_
+    static const int ReturnBuffPtrExtraSlot = 1;
+#else
+    static const int ReturnBuffPtrExtraSlot = 0;
+#endif
+
 #ifdef _TARGET_X86_
     UINT cbStackPop = argit.CbStackPop();
     pBuilder->WriteStackPop(cbStackPop / sizeof(TADDR));
 
     nStackSlots = nStackBytes / sizeof(TADDR) + NUM_ARGUMENT_REGISTERS;
 #else
-    nStackSlots = (sizeof(TransitionBlock) + nStackBytes - TransitionBlock::GetOffsetOfArgumentRegisters()) / TARGET_POINTER_SIZE;
+    nStackSlots = (sizeof(TransitionBlock) + nStackBytes - TransitionBlock::GetOffsetOfArgumentRegisters()) / TARGET_POINTER_SIZE + ReturnBuffPtrExtraSlot;
```

Comment:
I would like to add a new TransitionBlock::OffsetOfFirstGCRefMapSlot function as you've suggested offline instead. It seems it would be easier to read the code that way.